tracer/src/Datadog.Trace/Iast/SensitiveData/HeaderInjectionTokenizer.cs

```diff
@@ -0,0 +1,47 @@
+// <copyright file="HeaderInjectionTokenizer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Configuration;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.SensitiveData;
+
+/// <summary>
+/// We are already redacting sensitive values in the evidence (header values),
+/// but if the key of the returned header matches a regex, we should also redact the evidence value after ":"
+/// </summary>
+internal class HeaderInjectionTokenizer : ITokenizer
+{
+    private static Regex _keyPattern = new Regex(@"(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+    private static Regex _valuePattern = new Regex(@"(?i)(?:bearer\s+[a-z0-9\._\-]+|glpat-[\w\-]{20}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\w=\-]+\.ey[I-L][\w=\-]+(?:\.[\w.+/=\-]+)?|(?:[\-]{5}BEGIN[a-z\s]+PRIVATE\sKEY[\-]{5}[^\-]+[\-]{5}END[a-z\s]+PRIVATE\sKEY[\-]{5}|ssh-rsa\s*[a-z0-9/\.+]{100,}))", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
We should add a timeout to these regexes, otherwise they're vulnerable to DOS

Reply:
I have added some value as a temporary solution, but we should establish a common value for all the iast redaction regex. I have opened a task here: https://datadoghq.atlassian.net/browse/APPSEC-51099?atlOrigin=eyJpIjoiMTIxM2IzNzAzYzgyNDgxNmE4MjZhMjM3NTUzYWQzNzQiLCJwIjoiaiJ9

Thanks for pointing this!